docs/csharp/whats-new/version-update-considerations.md

```diff
@@ -9,15 +9,15 @@ ms.date: 09/19/2018
 
 Compatibility is a very important goal as new features are added to the C# language. In almost all cases, existing code can be recompiled with a new compiler version without any issue.
 
-More care may be required when you adopt new language features in a library. You may be creating a new library with features found in the latest version and need to ensure apps built using previous versions of the compiler can use it. Or you may be upgrading an existing library and many of your users may not have upgraded versions yet. As you make decisions on adopting new features, you'll need to consider two variations of compatibility: source-compatible and binary-compatible.
+More care may be required when you adopt new language features in a library. You may be creating a new library with features found in the latest version and need to ensure apps built using previous versions of the compiler can use it. Or you may be upgrading an existing library and many of your users may not have upgraded versions yet. As you make decisions on adopting new features, you'll need to consider two variations of compatibility: source compatibility and binary compatibility.
 
 ## Binary-compatible changes
 
-Changes to your library are **binary-compatible** when your updated library can be used without rebuilding applications and libraries that use it. Dependent assemblies are not required to be rebuilt, nor are any source code changes required.
+Changes to your library are **binary-compatible** when your updated library can be used without rebuilding applications and libraries that use it. Dependent assemblies don't need to be rebuilt, and no source code changes are required.
 
 ## Source-compatible changes
 
-Changes to your library are **source-compatible** when applications and libraries that use your library do not require source code changes, but the source must be recompiled against the new version to work correctly.
```

Comment:
@BillWagner This definition seemed incorrect to me - a source-incompatible change is one that appears *if* you recompile. A binary-incompatible change is where the source must be recompiled.

Reply:
Maybe we need to update the wording, but the current language is correct for the C# language definition of "source breaking" and "binary breaking" change.  If a change is "source compatible", code that compiled in version N still compiles in version N + 1. The example on line 47 shows an example. You must recompile. If you don't, you're app won't run using the new libraries. It's a binary break. But, if you recompile, your code adapts to the library change without any edits to the source.

Reply:
@gewarren 

I see your point. This article probably needs a fair amount of rework now. I first wrote this to address specific feedback where customers were choosing a newer language version while targeting an older TFM (typically .NET Standard 2.0). That's not a supported scenario.

Today, customers choose a language version / TFM as a pair. That should change the focus to be more about asking what changes a developer can make that are:

- compatible API changes
- binary breaking, but source compatible API changes
- source breaking API changes

I haven't thought it through, but my first impression is that a source breaking, but binary compatible API change can't exist. 

Those changes feel like scope creep for this PR. Do you want to make the edits, or open an issue and I'll address it after break?

Reply:
> I haven't thought it through, but my first impression is that a source breaking, but binary compatible API change can't exist.

You could have a new obsoletion as an error (instead of a warning) - that would be source incompatible but binary compatible.

I'll open a separate issue for this article then. Thanks Bill!